### DISCUSSION TITLE
Livewire 3

### DISCUSSION THREAD

**Discussion by lrljoe** [rappasoft/laravel-livewire-tables#1278] [General]
We are currently in the final stages of testing a Livewire 3 release.  More news to come soon!

We have an initial Beta out now
Please see
[https://github.com/rappasoft/laravel-livewire-tables/discussions/1305](https://github.com/rappasoft/laravel-livewire-tables/discussions/1305) for more details on changes etc

**Comment by Loydtafireyi** [comment]
This would be great

**Comment by lrljoe** [comment]
A quick update.  We have e been making improvements in various areas of the package, as the opportunity to make larger-scale changes rarely presents itself!

Rest assured that the package will still have all of the features that you find useful, with a few extras, and some performance improvements to boot.

**Reply by misodrobny** [reply]
This is awesome. Any idea when will be release for Livewire v3? I'm starting new project and I'd love to use new version.

**Reply by lrljoe** [reply]
Funnily enough @rappasoft and I were just discussing this, I think this week we'll stop any new functionality/improvements, and just finish off what's currently there.

From testing, there's a few minor bugs outstanding.  I genuinely think we'll have a beta out in the next week or two.

However, I'm also very conscious of the large number of bugs in Livewire v3 itself, and they could easily introduce a breaking change without notice.

**Reply by lrljoe** [reply]
It's worth noting, that from my testing, the number of breaking changes between v2 and v3 are relatively few in number, with very minor adjustments needed to any existing tables to get them to work in the new version.

We both worked hard to try and minimise the work it'd take for anyone to migrate to the new version.

**Reply by lrljoe** [reply]
v3.1.0 is released as stable with Livewire v3 support.

**Comment by lrljoe** [comment]
In some positive news, I believe I've reached the point where I'm just testing and doing very minor bug fixes.  There's nothing big left on my list for v3.x

Hopefully will be able to give an ETA for the beta by the end of this weekend.  It will almost certainly remain in beta until Livewire releases their stable version of 3.0

**Comment by lrljoe** [comment]
And we are down to a small handful of bugs, and tweaks.

These relate to improvements made in performance, or previous requests from the issues/discussions/prs, such as fixing table striping on bootstrap, allowing multiple joins to the same table, cursor pagination, a couple of new filters and columns, and the list goes on

Then it'll be another round or two of testing before the beta is ready.  I'm happy with the core functionality at this point.  It is some edge cases that I need to dig into and make sure are all working, as I don't want anyone having a painful time upgrading!

**Comment by lrljoe** [comment]
We edge ever closer to a beta release, especially with Livewire gradually reducing the number of open bugs.

We are now down to just a couple of minor bugs to address, some tests and docs to write.

I think we may release a beta before finishing all the docs and tests, but unsure!

**Comment by lrljoe** [comment]
As we get even closer - if anyone would like to volunteer to beta test, then please drop a note in the discord (preferably) or on here.

As it'll be a beta (as is Livewire 3 still) strongly recommend not using it in a production environment.

**Comment by alphaolomi** [comment]
Snr. Dev here how can I help with the testing of beta releases? Any notes of the latest communications?

**Reply by lrljoe** [reply]
So I'm presently working back through the list of changed functionality, as we've made some sweeping improvements across the board that have been on a back-burner for ages.

I think the next step will be a public demo/preview of V3, and then we'll go to a limited beta.

I'm planning to create a private "beta" channel on the Discord, and then granting some access to the v3 repository initially via there, but waiting for Anthony's go-ahead for that approach.

Effectively, rather than just fixing it for Livewire 3, which was more complex than we'd have liked anyway, so we ended up adding in a bunch of improvements at the same time, due to it being a major version, and thus won't just break existing components.
What we have tried to do though is make the migration as simple as possible, with **very** few major changes to the approach and methods.

**Reply by alphaolomi** [reply]
How do I join the project Discord?

**Reply by lrljoe** [reply]
[https://discord.gg/mkmEfq7hUC](https://discord.gg/mkmEfq7hUC).

We only very recently launched it, so don't expect to get spammed massively!

**Reply by alphaolomi** [reply]
Noted and thanks!

**Comment by lrljoe** [comment]
A beta should be available within the next 48 hours.  I am doing some final tests, but we'll then make it available for general testing.

Please note that, like Livewire core, we do not recommend using this in a production environment just yet!

**Comment by lrljoe** [comment]
Apologies for the delay, just triple-checking everything.  You may notice that there are new branches created, however these do not yet contain the v3 code!

I'm fully anticipating getting the branches updated this weekend, so that eager testers can use the dev branches to test, then we'll go to a beta release early next week most likely.

**Comment by Loydtafireyi** [comment]
Livewire V3 is now official/stable https://github.com/livewire/livewire/releases/tag/v3.0.0

**Reply by Loydtafireyi** [reply]
Is the beta release coming anytime soon/you're still far from making it a reality?

**Reply by lrljoe** [reply]
I'm expecting to move the v3 code across into the v3 branches this evening.

After a number of rounds of development, testing, enhancements etc!

A proper beta release will then follow, potentially at the weekend, but the dev branches will at least give people a chance to take a look!

**Reply by Loydtafireyi** [reply]
Thanks for the update, excited

**Reply by lrljoe** [reply]
You should be able to use the "v3-develop" branch now.

Please jump on the discord to ping me if anything is immediately seeming odd!

**Comment by lrljoe** [comment]
And you can now use the beta on the develop channel (v3-develop).

**Comment by trippo** [comment]
When do you plan to release a stable V3?

**Reply by lrljoe** [reply]
Official beta release (rather than using dev branches) will be in the next day or two.  

Then official release will probably be a week after that, bug dependent.  Docs are almost all finished too.

Two or three remaining fixes/changes to review and merge in.

**Comment by lrljoe** [comment]
As an update, all development is now merged into v3-master

Just waiting for docs to go live, then a proper beta release will be out, hopefully shortly followed by a full release.

Don't want to do an official beta without docs being live!

Proper release will be dependent on there only being edge case issues present, or ideally no issues at all (one can dream)

**Comment by lrljoe** [comment]
Beta for v3 is now out, so you can use the standard method for including the package.

Note that you may need to adjust your minimum stability until it comes out of beta.

**Reply by nathan-io** [reply]
Great news! Any idea on timeframe for the upgrade guide and documentation for the new Livewire component column type?

**Reply by lrljoe** [reply]
Good questions!  

In terms of upgrade guide, there is minimal difference to the behaviour of v2, but I do plan to get it finished off this weekend.

The key differences are

The field names returned to the reorder function have changed (they are now the same as the actual db fields), but I do plan to add a toggle for this behaviour to smooth any transition, prior to an official release.

For search, ability to setSearchLive and  setSearchBlur.

In terms of the LivewireComponent aspects (Filter and Column), I'm hoping to have those documented this weekend.

Most of the work has been behind the scenes and non-impacting, in so far as possible.

The big change will be if you've published / modified any views, as a lot of them have been massively rewritten to improve behaviour/performance/ease of maintenance.

**Comment by lrljoe** [comment]
beta 003 is now out.

I am continuing to work on the upgrade docs, but there are very minimal breaking changes, other than the requirement to remove published views if you have done so on the past

**Comment by lrljoe** [comment]
Aaaand v3.1.0 is out as a stable release.
